### PR TITLE
Add property to filter scores by entity kind

### PR DESCRIPTION
**Pull request by T0RAT0RA**
Add property to filter scores by entity kind.

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by T0RAT0RA** [comment]
I'm not sure how to test the changes yet, as the filtering happens in the `ScoringDataApi` which is always mock in the tests.
I'll think about it.

File: plugins/score-card/src/api/ScoringDataJsonClient.ts [hunk 1]
```diff
@@ -72,10 +72,10 @@ export class ScoringDataJsonClient implements ScoringDataApi {
     return this.extendEntityScore(result, undefined);
   }
 
-  public async getAllScores(): Promise<EntityScoreExtended[] | undefined> {
+  public async getAllScores(entityKinds?: ScorableEntityKind[]): Promise<EntityScoreExtended[] | undefined> {
```
Comment (jvilimek) [inline plugins/score-card/src/api/ScoringDataJsonClient.ts:75]: just a thought... how about `entityKindFilter`? Have you considered this as well?

**Review by jvilimek — APPROVED** [review]
Thanks for contribution as always! Looking good, few (not so much important) questions though.

**Comment by jvilimek** [comment]
> EDIT: I'll just create a new `ScoringDataJsonClient.test.ts` test file

That would be awesome!

File: plugins/score-card/src/api/types.ts [hunk 2]
```diff
@@ -59,3 +59,11 @@ export interface EntityScoreExtended extends EntityScore {
   reviewer: CompoundEntityRef | undefined;
   reviewDate: Date | undefined;
 }
+
+
+export enum ScorableEntityKind {
```
[the remark this answers is not in the document]
Comment (T0RAT0RA) [reply plugins/score-card/src/api/types.ts:64]: The catalog doesn't provide an enum I could reuse, they're set in the different interfaces:
Ex: 
```ts
declare interface SystemEntityV1alpha1 extends Entity {
    apiVersion: 'backstage.io/v1alpha1' | 'backstage.io/v1beta1';
    kind: 'System';
    spec: {
        owner: string;
        domain?: string;
    };
}
```

I could leave it a string if you prefer.

File: plugins/score-card/src/api/ScoringDataJsonClient.ts (hunk 1 above)
Comment (T0RAT0RA) [reply plugins/score-card/src/api/ScoringDataJsonClient.ts:75]: Haven't consider it.
It's is more explicit :+1:

**Comment by T0RAT0RA** [comment]
I'm not able to find a way to test the `ScoringDataJsonClient`, I don't find good documentation  on testing Backstage API.

Here is what I have so far (wich is not working):

```ts
  it('should filter entities by kind', async () => {
    const mockConfig = new MockConfigApi({
      app: { baseUrl: 'https://example.com' },
    });
    const mockFetch = new MockFetchApi();

    const api = new ScoringDataJsonClient({
      configApi: mockConfig,
      fetchApi: mockFetch,
      catalogApi: catalogApiRef as AnyApiRef
    });

    const entities = await api.getAllScores();
    expect(entities).toBe({});
  });
```

I'm looking for a way to mock the `catalogApi`.

File: plugins/score-card/src/api/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jvilimek) [reply plugins/score-card/src/api/types.ts]: Perfect, closing!

File: plugins/score-card/src/api/ScoringDataJsonClient.ts (hunk 1 above)
Comment (jvilimek) [reply plugins/score-card/src/api/ScoringDataJsonClient.ts]: Thanks, closing!

**Comment by jvilimek** [comment]
> ```ts
> MockConfigApi
> ```

I am not a typescript nor Backstage guru, but what I have found is this: [GoCdBuildsComponent.test.tsx](https://github.com/backstage/backstage/blob/master/plugins/gocd/src/components/GoCdBuildsComponent/GoCdBuildsComponent.test.tsx) and this [OwnershipCard.test.tsx](https://github.com/backstage/backstage/blob/master/plugins/org/src/components/Cards/OwnershipCard/OwnershipCard.test.tsx) which seems to do what you want to achieve?

**Comment by T0RAT0RA** [comment]
Thanks @jvilimek, exactly what I needed, I'll write the tests this week.

**Review by jvilimek — APPROVED** [review]

**Comment by jvilimek** [comment]
Perfect! Thanks a lot for your contribution!

**Pull request merged**